runtime/ts/storage/firebase-storage.ts

```diff
@@ -147,11 +147,8 @@ baseStorageKey(type: Type, keyString: string): string {
     return new FirebaseKey(s);
   }
 
-  // referenceMode is only referred to if shouldExist is false, or if shouldExist is 'unknown'
-  // but this _join creates the storage location. 
-  async _join(id: string, type: Type, keyString: string, shouldExist: boolean | 'unknown', referenceMode = false) {
-    assert(!type.isVariable);
-    assert(!type.isCollection || !type.primitiveType().isVariable);
+  // Exposed for SyntheticStorage. TODO: better API
+  attach(keyString: string) : {fbKey: FirebaseKey, reference: firebase.database.Reference} {
```

Comment:
Add a comment that this is only here to share the this.apps list and that we will eventually refactor arc storage such that it's sufficiently variable-like that we can access the manifest using a standard FirebaseVariable.

Reply:
Done.